aten/src/ATen/record_function.h

```diff
@@ -299,6 +299,31 @@ struct TORCH_API RecordFunction {
     before(fn, current_sequence_nr);
   }
 
+  template <typename F>
+  void before(
+      F fn,
+      c10::ArrayRef<const c10::IValue> args,
+      const std::unordered_map<std::string, IValue>* kwargs,
+      int64_t current_sequence_nr = -1) {
+    if (!isActive()) {
+      return;
+    }
+    kwinputs_ = *kwargs;
+    before(std::move(fn), args, current_sequence_nr);
+  }
+
+  template <typename F>
+  void before(
+      F fn,
+      const std::unordered_map<std::string, IValue>* kwargs,
+      int64_t current_sequence_nr = -1) {
+    if (!isActive()) {
+      return;
+    }
+    kwinputs_ = *kwargs;
+    before(fn, current_sequence_nr);
+  }
```

Comment:
is this variant needed? it seems like it's not used in the macro that you added

Reply:
It is because in this line https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/record_function.h#L540
needsInputs() can be false but we have kwargs as the first variable in ##__VA_ARGS__ so we need to abide by a strict template that has kwargs and a sequence_num in that order. 

I was thinking since this template is only used in this case today, that we could just ignore kwargs and not bother storing it, but then thought that in the future the template might be used for other instances in which we want to save the kwargs.